diesel/src/sqlite/connection/app_defined_fn.rs

```diff
@@ -0,0 +1,69 @@
+//! Plumbing for the implementation of the application defined function API
+
+use std::os::raw as libc;
+
+use super::ffi;
+use super::into_sqlite_result::IntoSqliteResult;
+use super::from_sqlite_value::FromSqliteValue;
+
+/// Context is a wrapper for the SQLite function evaluation context.
+#[derive(Debug)]
+pub struct Context<'a> {
+    ctx: *mut ffi::sqlite3_context,
+    args: &'a [*mut ffi::sqlite3_value],
+}
+
+// Context is translated from rusqlite
+impl<'a> Context<'a> {
+    /// Returns the number of arguments to the function.
+    pub fn len(&self) -> usize {
+        self.args.len()
+    }
+
+    /// Returns `true` when there is no argument.
+    pub fn is_empty(&self) -> bool {
+        self.args.is_empty()
+    }
+
+    /// Returns the `idx`th argument as a `T`.
+    ///
+    /// # Failure
+    ///
+    /// Will panic if `idx` is greater than or equal to `self.len()`.
+    pub fn get<T>(&self, idx: usize) -> T
```

Comment:
It's atypical Rust to have a `get` method panic; it should return an Option, or you should `impl Index for Context`

Reply:
In that case, I'm partial to `impl Index`, since `Context` is essentially a wrapper around an array.